### PR TITLE
Fix lockyloo golang

### PR DESCRIPTION
**Pull request by rdallman**
source: https://github.com/mattes/migrate/pull/299

got wind that this is the new happening place for development

Quick summary:

*   mysql driver relies on >=go1.9 now
*   mysql driver now uses same conn for migrations to fix GET_LOCK & RELEASE_LOCK behavior
*   upped time out to 10s on GET_LOCK, 1s timeout can too easily leave us in a
    state where we think we have the lock but it has timed out (during the
    operation).
*   fixes SetVersion which was not using the tx it was intending to, and fixed a
    bug where the transaction could have been left open since Rollback or Commit
    may never have been called.

thanks for taking over maintenance of this project.

### CODE REVIEW

File: database/mysql/mysql.go [hunk 1]
```diff
@@ -162,9 +172,9 @@ func (m *Mysql) Lock() error {
 		return err
 	}
 
-	query := "SELECT GET_LOCK(?, 1)"
+	query := "SELECT GET_LOCK(?, 10)"
```
Comment (dhui) [inline database/mysql/mysql.go]: We should probably make the timeout configurable, but I think it's fine to make that change in a separate PR

**Comment by rdallman** [comment]
go 1.10 is released bump :) woohoo, tests are fast now

**Comment by dhui** [comment]
Fixed in release [v3.2.0](https://github.com/golang-migrate/migrate/releases/tag/v3.2.0)
